### DISCUSSION TITLE
lazer should ask whether to be the default client for opening .osz files

### DISCUSSION THREAD

**Discussion by kentpz727** [ppy/osu#30161] [Ideas]
I realized this was an issue I had when I watched the most recent development video, and that was when my beatmaps started opening in lazer by default rather than on stable, which I preferred at the time and still do. While I did eventually figure it out and changed it back to stable, it confused me at the start, and it didn't help that both clients were the same name in the Default Apps setting for a while until recently. This was an annoying inconvenience at best, and a very confusing few hours of troubleshooting at worst, and it should definitely be looked into.

I think the behavior for this should be opt in rather than be set automatically on install, similar to how browsers would ask to make themselves the "Default Browser". osu!lazer should ask on initial setup whether to make it the default client for .osz files, or keep it as it is with files continuing to stable. That is of course, if there even is a stable client detected, otherwise it doesn't ask as to not overwhelm new players.

Maybe add a toggle or a button somewhere in the options for this as well? Just my two cents.

**Comment by smallketchup82** [comment]
This was discussed to an extent in #29603, and my stance on this subject hasn't really changed.

If you *explicitly* select stable to be the handler for osu related files (via the "open with" dialogs, or default apps section in settings), lazer won't do anything except register itself as an additional option to handle them. If you haven't manually selected any program (and left it up to stable or lazer to decide), lazer will take precedent over stable and begin handling the files. Meaning users will only ever have to configure these once and never again (unless they intend to switch clients).

The only other user friendly solution is to have it be opt-in if lazer ever gets a setup wizard (NSIS, WiX, etc), as I want to avoid adding more in-game settings. Until that day comes, I believe the current approach is best and I personally wont go out of my way to change it.
